org.eclipse.debug.core/core/org/eclipse/debug/internal/core/LaunchConfigurationWorkingCopy.java

```diff
@@ -278,9 +278,10 @@ private void doSave0(IProgressMonitor monitor) throws CoreException {
 	 * @exception CoreException if writing the file fails
 	 */
 	protected void writeNewFile(IProgressMonitor monitor) throws CoreException {
+		String lineDelimeter = ResourcesPlugin.getLineSeparator(getFile());
```

Comment:
For new file, getFile() will be null so will eventually use System.lineSeparator() ? How will the configured workspace preference be used?

Reply:
Good point. I think `getContainer().getProject().getDefaultLineSeparator()` should work in this case, otherwise if container is null we assume it's outside the workspace and use System.lineSeparator()? Do you think this is ok?

Reply:
So, added getLineSeparator() in LaunchConfiguration.java that calls getFile().getLineSeparator(true) that already checks for the presence of a container resource. It also check project/instance scope automatically.

If it returns null, the destinations is either workspace metadata or outside workspace, hence system separator is used.

Reply:
Thanks , works well.